### DISCUSSION TITLE
Spanish support in RealtimeSTT

### DISCUSSION THREAD

**Discussion by garlandwong** [KoljaB/RealtimeSTT#149] [Q&A]
Does anyone know how to enable spanish support in realtimeSTT?

**Comment by KoljaB** [comment]
Call AudioToTextRecorder with parameter language="es".

**Reply by garlandwong** [reply]
thank you for the help.  I am getting the following error:
RealTimeSTT: faster_whisper - WARNING - The current model is English-only but the language parameter is set to 'es'; using 'en' instead.
the funny thing is it works.  What model do you recommend?

**Reply by KoljaB** [reply]
Any model that does not end with ".en"
For example medium or large-v2. Depends on if you prefer speed or transcription quality.

**Reply by homelab-00** [reply]
>What model do you recommend?

Depends on your hardware. With a dedicated Nvidia GPU you can probably run the biggest models. For AMD cards you need to use ROCm but that's only available for Linux. If you only have a CPU medium or base model should be best.

Of course they should be the multilingual version and not the .en one, as KoljaB said.

Also keep in mind that you can use any model in CTranslate2 format from Hugging Face. I'm using `'model': 'Systran/faster-whisper-large-v3'` (from https://huggingface.co/Systran/faster-whisper-large-v3). Works fine for Greek and English.
Note that distil versions might auto translate non English speech into English (depends on the language I think).

**Comment by KoljaB** [comment]
If it works although complaining about english model it might be that the realtime transcription model is at "tiny.en" and the main transcription model at "large-v2". Then you would not really get spanish realtime transcription (possibly translated to english) but you'd get a spanish final transcription.

**Reply by garlandwong** [reply]
For some reason.  The spanish translation is not working anymore.  I am getting the following warning now.  This used to work.
all right, come aside.RealTimeSTT: faster_whisper - WARNING - The current model is English-only but the language parameter is set to 'es'; using 'en' instead.

**Reply by KoljaB** [reply]
Please check your model and realtime_model_type parameters, one of them ends on ".en"
Remove the ".en" there and you should be good.
